docs/sphinx/_toc.yml.in

```
@@ -0,0 +1,23 @@
+defaults:
+  numbered: False
+root: src/index.rst
+subtrees:
+  - entries:
+    - file: src/getting-started/getting-started
+  - caption: Guides
+    entries:
+    - file: src/contributor-guide
+    - file: src/developer-guide
+  - caption: Concepts
+    entries:
+    - file: src/concepts/benchmarking.rst
+    - file: src/concepts/tuning.rst
+    - file: src/concepts/kernels.rst
+    - file: src/concepts/library-creation.rst
+  - caption: Reference
+    entries:
+      - file: src/api-reference/api-reference.rst
+      - file: src/cli-reference/cli-reference.rst
+  - entries:
+    - file: license.rst
+
```

The following will allow us to have collapsable sections but in some cases we will need to introduce files where there was previously captions:

```suggestion
defaults:
  numbered: False
root: src/index.rst
subtrees:
  - entries:
    - file: src/getting-started.rst
    - file: src/concepts.rst
      entries:
        - file: src/concepts/benchmarking.rst
        - file: src/concepts/tuning.rst
        - file: src/concepts/kernels.rst
        - file: src/concepts/library-creation.rst
    - file: src/api-reference.rst
    - file: src/cli-reference.rst
    - file: src/contributor-guide.rst
    - file: src/developer-guide.rst
    - file: license.rst
    
```

---

Note that each of the outer most sections e.g. concepts, getting started, cli reference, will have a corresponding subdirectory where we can place the subsections that belong.